src/transformers/trainer_tf.py

```diff
@@ -218,6 +270,12 @@ def evaluate(
 
         output = self._prediction_loop(eval_ds, description="Evaluation")
 
+        logs = {**output.metrics}
+        # add epoch when running evaluate from training loop
+        if self.epoch_logging is not None:
```

Comment:
Here I made a mistake and it should be `if hasattr(self, "epoch_logging")`.

An easier solution (if we move `epoch_logging` to `__init__`) would be to avoid this verification at all since this attribute will always exist (and there's no harm in using an epoch of `0` when doing no training).

Let me know your preference.

Reply:
You should not log anything here. That's why there is the logging_step and eval_step parameters. Then the evaluate method should stay as it was. And log only in the train metrhod.

Reply:
No epoch should not appear in the logs even when you do only eval.

Reply:
See my next comment. It is hard to identify when I'm in `eval/predict` only and there could be some issues when wanting to eval prior training.